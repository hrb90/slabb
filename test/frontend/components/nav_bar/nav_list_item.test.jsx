import React from 'react';
import NavListItem from '../../../../frontend/components/nav_bar/nav_list_item';
import renderer from 'react-test-renderer';

const mockNameFix = jest.fn();
mockNameFix.mockReturnValue("Fixed!");

const mockFetchChannel = id => jest.fn().mockReturnValue("Fetched!");

const basicChannel = {
      id: 2,
      name: "channel",
      newMessages: false,
      channel_type: "channel"
    };

const newMessagesChannel = {
      id: 2,
      name: "channel",
      newMessages: true,
      channel_type: "channel"
    };

const dmChannel ={
      id: 2,
      name: "channel",
      newMessages: false,
      channel_type: "dm"
    };

describe('NavListItem', () => {
  it('renders basic props correctly', () => {
    const tree = renderer.create(
      <NavListItem channel={ basicChannel } currentChannelId={ 1 }
        fixDMName={ mockNameFix } fetchChannel={ mockFetchChannel } />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders new messages correctly', () => {
    const tree = renderer.create(
      <NavListItem channel={ newMessagesChannel } currentChannelId={ 1 }
        fixDMName={ mockNameFix } fetchChannel={ mockFetchChannel } />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders the current channel correctly', () => {
    const tree = renderer.create(
      <NavListItem channel={ basicChannel } currentChannelId={ 2 }
        fixDMName={ mockNameFix } fetchChannel={ mockFetchChannel } />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('calls fixDMName when passed a DM channel', () => {
    renderer.create(
      <NavListItem channel={ dmChannel } currentChannelId={ 1 }
        fixDMName={ mockNameFix } fetchChannel={ mockFetchChannel } />
    );
    expect(mockNameFix).toBeCalled();
  });
});
