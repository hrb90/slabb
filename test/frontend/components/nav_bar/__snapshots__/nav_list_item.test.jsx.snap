// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavListItem renders basic props correctly 1`] = `
<li
  className="nb-channel"
  onClick={[Function]}
>
  #channel
</li>
`;

exports[`NavListItem renders new messages correctly 1`] = `
<li
  className="nb-channel nb-channel-bold"
  onClick={[Function]}
>
  #channel
</li>
`;

exports[`NavListItem renders the current channel correctly 1`] = `
<li
  className="nb-channel current-nb-channel"
  onClick={[Function]}
>
  #channel
</li>
`;
